odc/geo/gridspec.py

```diff
@@ -82,13 +95,15 @@ def dimensions(self) -> Tuple[str, str]:
     @property
     def alignment(self) -> Tuple[float, float]:
         """Pixel boundary alignment."""
-        y, x = (orig % abs(res) for orig, res in zip(self.origin, self.resolution))
+        y, x = (
+            orig % abs(res) for orig, res in zip(self.origin.yx, self.resolution.yx)
+        )
         return (y, x)
 
     @property
     def tile_shape(self) -> Tuple[int, int]:
         """Tile shape in pixels (Y,X order, like numpy)."""
```

Comment:
Maybe? kinda on the fence about `.shape`, I feel like for `.shape` there is no ambiguity, so adding extra indirection for interfacing to ndarray might be annoying/confusing. Maybe it's worth creating `Shape(XY[int])` and implementing `__iter__` on that so that you could pass it in to `numpy` functions where `shape=` is expected, while still being able to use all the other `XY` methods too.